cmake/utils.cmake

```diff
@@ -10,3 +10,13 @@ macro(add_dir DIRS)
     list( APPEND ${PROJECT_NAME}__SOURCES ${${dir}__SOURCES_C} ${${dir}__SOURCES_CPP} )
   endforeach()
 endmacro()
+
+
+# ------------------------------------------
+# Detect arch type ( x86 or x64 )
+# ------------------------------------------
+if (CMAKE_SIZEOF_VOID_P EQUAL 8)
+  set(ARCH_TYPE x64)
+else(CMAKE_SIZEOF_VOID_P EQUAL 4)
+  set(ARCH_TYPE x86)
```

Comment:
Will it properly work for non-x86 archs (ARM/ARM64/E2K)?

Reply:
Yes it should work, but for sure better to check it out, i have an Raspberry Pi 3, but i can check only on next weekend's. Would be great if someone who have that types of processor architecture take a test.